lua/wikis/commons/ThisDay.lua

```diff
@@ -0,0 +1,385 @@
+					}
+				)
+			end
+
+			return line
+		end)
+
+		return ThisDay._buildListWidget(lines)
+	end
+end
+
+--- Get and display patches that happened on a given date (falls back to today)
+---@param args {date: string?, month: string|integer|nil, day: string|integer|nil}
+---@return string|Widget?
+function ThisDay.patch(args)
+	local patchData = Query.patch(ThisDay._readDate(args))
+
+	if not patchData then
+		return 'There were no patches on this day'
+	else
+		local lines = Array.map(patchData, function (patch)
+			local patchYear = patch.date:sub(1, 4)
+			return {
+				HtmlWidgets.B{
+					children = {patchYear}
+				},
+				': ',
+				Link{link = patch.pagename, children = patch.name},
+				' released'
+			}
+		end)
+
+		return ThisDay._buildListWidget(lines)
+	end
+end
+
+--- Get and display tournament wins that happened on a given date (falls back to today)
+---@param args {date: string?, month: string|integer|nil, day: string|integer|nil}
+---@return string|Widget?
+function ThisDay.tournament(args)
+	local tournamentWinData = Query.tournament(ThisDay._readDate(args))
+
+	if not tournamentWinData then
+		return 'No tournament ended on this date'
+	else
+		local _, byYear = Array.groupBy(tournamentWinData, function(placement) return placement.date:sub(1, 4) end)
+
+		local display = {}
+		for year, yearData in Table.iter.spairs(byYear) do
+			Array.appendWith(display,
+				HtmlWidgets.H4{
+					children = { year }
+				},
+				'\n',
+				ThisDay._displayWins(yearData)
+			)
+		end
+		mw.logObject(display)
+		return HtmlWidgets.Fragment{children = display}
+	end
+end
+
+--- Display win rows of a year
+---@param yearData placement[]
+---@return Widget?
+function ThisDay._displayWins(yearData)
+	local display = Array.map(yearData, function (placement)
+		local displayName = placement.shortname
+		if String.isEmpty(displayName) then
+			displayName = placement.tournament
+			if String.isEmpty(displayName) then
+				displayName = string.gsub(placement.pagename, '_', ' ')
+			end
+		end
+
+		local row = {
+			LeagueIcon.display{
+				icon = placement.icon,
+				iconDark = placement.icondark,
+				link = placement.pagename,
+				date = placement.date,
+				series = placement.series,
+				name = placement.shortname,
+			},
+			' ',
+			Link{ link = placement.pagename, children = displayName },
+			' won by '
+		}
+
+		local opponent
+		if placement.opponenttype then
+			opponent = Opponent.fromLpdbStruct(placement)
+
+		-- legacy opponent building
+		elseif placement.mode == Config.soloMode then
+			opponent = {
+				type = Opponent.solo,
+				players = {
+					displayName = placement.extradata.participantname or placement.participant,
+					flag = Flags.CountryName(placement.participantflag),
+					pageName = placement.participant,
+				},
+			}
+		else
+			opponent = {
+				type = Opponent.team,
+				name = placement.participant,
+				template = placement.participanttemplate or placement.participant:lower():gsub('_', ' '),
+			}
+		end
```

Comment:
all prize pools should be converted by now
```suggestion
		local opponent = Opponent.fromLpdbStruct(placement)
```

Reply:
they have local overwrites anyways ... (using local `Module:This day` via their local templates ...)
those 2 wikis are so far from standard ...